### PR TITLE
add manual log methods to highlight go sdk

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Enhances the go sdk with manual log functions that can be used without a logging library.

## How did you test this change?

Local deploy.
Screenshot 2023-03-08 at 2 16 06 PM

## Are there any deployment considerations?

Will release new highlight-go tag.

### CODE REVIEW

File: sdk/highlight-go/highlight.go [hunk 1]
```diff
@@ -255,6 +255,20 @@ func SetProjectID(id string) {
 	projectID = id
 }
 
+func SetLogOut(print bool) {
+	logOut = print
```
Comment (et) [inline sdk/highlight-go/highlight.go:259]: I don't see why anyone would want this. If anything, this should be setting the min log level. For example, here's how logrus does it:
https://github.com/sirupsen/logrus#level-logging

File: sdk/highlight-go/highlight.go [hunk 2]
```diff
@@ -255,6 +255,20 @@ func SetProjectID(id string) {
...
+}
+
+func Print(message string) {
```
Comment (et) [inline sdk/highlight-go/highlight.go:262]: Per the above, I would expect we'd set the config in the logging sdk, not here. Which would mean, we don't need this wrapper around `fmt.Print`

File: sdk/highlight-go/log/log.go [hunk 3]
```diff
@@ -52,6 +55,70 @@ type VercelLog struct {
...
+	var tagsStr string
+	if len(tags) > 0 {
+		s, _ := json.Marshal(tags)
+		tagsStr = fmt.Sprintf(" %s", s)
+	}
+	highlight.Print(fmt.Sprintf("[%s] %s%s\n", level, message, tagsStr))
+	LogWithContext(context.TODO(), level, message, tags...)
+}
+
+func Trace(message string, tags ...attribute.KeyValue) {
```
[the remark this answers is not in the document]
Comment (et) [reply sdk/highlight-go/log/log.go:102]: Alternatively, we could also provide a `WithContext` so one can do `hlog.WithContext(ctx).Info(....)`

File: sdk/highlight-go/highlight.go (hunk 1 above)
Comment (Vadman97) [reply sdk/highlight-go/highlight.go:259]: this came from a customer request to use highlight side by side with their existing logger that prints to console. i'll adjust this to customize a max level that logs will be emitted for.

File: sdk/highlight-go/log/log.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vadman97) [reply sdk/highlight-go/log/log.go:102]: i like that idea - will add a `WithContext` provider

File: sdk/highlight-go/highlight.go (hunk 2 above)
Comment (Vadman97) [reply sdk/highlight-go/highlight.go:262]: yup, makes sense to move this and the above logic to the `hlog` package

File: sdk/highlight-go/log/log.go [hunk 4]
```diff
@@ -52,6 +55,70 @@ type VercelLog struct {
 	Proxy       VercelProxy `json:"proxy"`
 }
 
+func HighlightTags(projectID, sessionID, requestID string) []attribute.KeyValue {
```
[the remark this answers is not in the document]
Comment (Vadman97) [reply sdk/highlight-go/log/log.go:58]: i'll be consistent with the public `WithContext` and expose this as an `hlog.WithSession(...).Log()` function to customers

File: backend/main.go [hunk 5]
```diff
@@ -228,14 +228,19 @@ func main() {
...
 		H.SetGraphqlClientAddress("https://localhost:8082/public")
-		H.SetOTLPEndpoint("http://collector:4318")
+		if util.IsInDocker() {
+			H.SetOTLPEndpoint("http://collector:4318")
+		}
 	}
 	H.Start()
 	defer H.Stop()
 	H.SetDebugMode(log.StandardLogger())
+	hlog.SetOutput(true)
```
Comment (et) [inline backend/main.go:241]: This isn't actually needed, correct?

File: backend/main.go [hunk 6]
```diff
@@ -228,14 +228,19 @@ func main() {
...
+	hlog.SetOutputLevel(hlog.DebugLevel)
```
Comment (et) [inline backend/main.go:242]: Nice!

**Review by et — APPROVED** [review]
Looks great!

File: backend/main.go (hunk 5 above)
Comment (Vadman97) [reply backend/main.go:241]: right, `true` is the default. keeping it in as an example though.

**Pull request merged**